vnext/Shared/Networking/OriginPolicyHttpFilter.cpp

```diff
@@ -443,21 +440,24 @@ OriginPolicy OriginPolicyHttpFilter::ValidateRequest(HttpRequestMessage const &r
 void OriginPolicyHttpFilter::ValidateAllowOrigin(
     hstring const &allowedOrigin,
     hstring const &allowCredentials,
-    IInspectable const &iRequestArgs) const {
+    IMap<hstring, IInspectable> props) const {
   // 4.10.1-2 - null allow origin
   if (L"null" == allowedOrigin)
     throw hresult_error{
         E_INVALIDARG,
         L"Response header Access-Control-Allow-Origin has a value of [null] which differs from the supplied origin"};
 
-  bool withCredentials = iRequestArgs.as<RequestArgs>()->WithCredentials;
+  bool withCredentials = props.Lookup(L"RequestArgs").as<RequestArgs>()->WithCredentials;
```

Comment:
>Lookup [](http://example.com/codeflow?start=31&length=6)

Why `Lookup` here and `TryLookup` later. 
 * TryLookup is not in the docs: [https://docs.microsoft.com/en-us/uwp/api/windows.foundation.collections.imap-2.lookup?view=winrt-22621](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.collections.imap-2.lookup?view=winrt-22621)
* Coming from C# that Try signature does not make any sense, it should return a boolean with out parameter for the value

Reply:
The general convention in C++/WinRT is:
- *SomeAction*: May throw
- *TrySomeAction*: Best effort. Will return a default/null value if it fails.

I use Lookup where the properties are always expected. It would literally be an exceptional case otherwise, and thrown accordingly.
TryLookup is used on conditions where it is possible that such property has not yet been set or will never be.